platform/darwin/src/MGLGeoJSONSource.mm

```diff
@@ -79,21 +80,43 @@ - (void)commonInit
     if (self.URL) {
         NSURL *url = self.URL.mgl_URLByStandardizingScheme;
         source->setURL(url.absoluteString.UTF8String);
-        _features = nil;
+        _feature = nil;
     } else if (self.geoJSONData) {
+        
         NSString *string = [[NSString alloc] initWithData:self.geoJSONData encoding:NSUTF8StringEncoding];
-        const auto geojson = mapbox::geojson::parse(string.UTF8String).get<mapbox::geojson::feature_collection>();
-        source->setGeoJSON(geojson);
-        _features = MGLFeaturesFromMBGLFeatures(geojson);
+        // TODO: Figure out a more performant way than serializing the data just to do a type check
+#warning - DEBUG CODE
+        id serialized = [NSJSONSerialization JSONObjectWithData:self.geoJSONData options:0 error:nil];
+        BOOL isFeatureCollection = [serialized isKindOfClass:NSArray.class] || ([serialized isKindOfClass:NSDictionary.class] && [[(NSDictionary *)serialized objectForKey:@"type"] isEqualToString:@"FeatureCollection"]);
+        
+        if (isFeatureCollection) {
```

Comment:
Why does this conditional exist? `mbgl::style::Source::setGeoJSON()` takes a `mapbox::geojson::geojson`, which is the type returned by `mapbox::geojson::parse()`, so no `get()` is needed for either case. `_feature` is of type `id <MGLFeature>`, so `MGLFeatureFromMBGLFeature()` works for both cases too.